lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -1,17 +1,34 @@
+import getpass
 import logging
 import os
+import shlex
 import shutil
+import subprocess
 import tempfile
 
 from galaxy import model
-from galaxy.model import store
+from galaxy.util import unicodify
 from galaxy.version import VERSION_MAJOR
 
 log = logging.getLogger(__name__)
 
 ATTRS_FILENAME_HISTORY = 'history_attrs.txt'
 
 
+def _chown(path, jeha, app, user):
```

Comment:
don't we have a function like that already somewhere in the codebase ? could we re-use it ?

Reply:
Yeah, at least here: https://github.com/galaxyproject/galaxy/blob/b91a205f89c906ce062abc6ae5bb7b5ac73ebfa8/lib/galaxy/tools/actions/upload_common.py#L320 and here https://github.com/galaxyproject/galaxy/blob/9662164ad68b39adf5a5606a7aa8e388f6a79f1e/lib/galaxy/jobs/__init__.py#L2169 these are currently bound to a class which we can't access from here as far as I see.

I was thinking about introducing a new function (doing popen, and error handling, and maybe pwent stuff) in util that could be reused. Would this be a good idea?

Reply:
Yes, I think that would be great.

Reply:
OK. Function added. Need to test on my setup. 

Will introduce this function at the other places later in a separate PR.

Currently the code is missing to call the chown also for archive_source in the import case. Would be nice, but I do not have an idea how to access the path here: https://github.com/galaxyproject/galaxy/blob/c2580727060583462abc76acec00577090c16da0/lib/galaxy/jobs/__init__.py#L1789

Reply:
Not sure about the import archive_source, that would probably be handled somewhere during upload ?

Could you also use the new function in this PR as well ? This avoids having slightly divergent functions scattered in the code, and is also a good way to figure out if anything is missing in the new function (which looks good!).